src/dex-ui/pages/dao/MultiSigDAODashboard/DepositTokensModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, Divider, Flex, Text } from "@chakra-ui/react";
+import { Color, FormDropdown, FormInput, NewTokenIcon } from "@dex-ui-components";
+import { TokenBalance, useAccountTokenBalances } from "@dex-ui/hooks";
+import { ChangeEvent } from "react";
+import { useForm } from "react-hook-form";
+import { HBARTokenId, HBARSymbol } from "@services";
+
+interface DepositTokensModalBodyProps {
+  safeId: string;
+  handleDepositClicked: (data: DepositTokensFormData) => void;
+  handleCancelClicked: () => void;
+  handleAssociateTokenClicked: () => void;
+}
+
+export interface DepositTokensFormData {
+  tokenId: string;
+  amount: number;
+  decimals: number;
+}
+
+const DepositTokensModal = (props: DepositTokensModalBodyProps) => {
+  const { safeId, handleDepositClicked, handleCancelClicked, handleAssociateTokenClicked } = props;
+  const {
+    handleSubmit,
+    register,
+    setValue,
+    formState: { errors },
+  } = useForm<DepositTokensFormData>();
+
+  const accountTokenBalancesQueryResults = useAccountTokenBalances(safeId ?? "");
+  const { data: tokenBalances } = accountTokenBalancesQueryResults;
+  const assetDropdownOptions =
+    tokenBalances?.map((asset: TokenBalance) => {
+      const { symbol, tokenId } = asset;
+      if (symbol === HBARSymbol) {
+        return {
+          label: symbol,
+          value: HBARTokenId,
+        };
+      }
+      return {
+        label: symbol,
+        value: tokenId,
+      };
+    }) ?? [];
+
+  async function onSubmit(data: DepositTokensFormData) {
+    handleDepositClicked(data);
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Flex direction="column" gap={4}>
+        <Divider marginX={-5} paddingX={5} />
+        <FormDropdown
+          label="Assets"
+          placeholder="Select a token"
+          data={assetDropdownOptions}
+          isInvalid={Boolean(errors?.tokenId)}
+          errorMessage={errors?.tokenId && errors?.tokenId?.message}
+          register={register("tokenId", {
+            required: { value: true, message: "A token is required." },
+            onChange: (e: ChangeEvent<HTMLSelectElement>) => setValue("tokenId", e.target.value),
+          })}
+        />
+        <FormInput<"amount">
+          inputProps={{
+            id: "amount",
+            label: "Amount",
+            type: "number",
+            placeholder: "Enter amount",
+            register: {
+              ...register("amount", {
+                required: { value: true, message: "An amount is required." },
+              }),
+            },
```

Comment:
We could reuse new` TokenAmountInput` component instead of `FormInput` same as `Send` token screen